### PR TITLE
New endpoints structure

### PR DESCRIPTION
**Pull request by trojkat**
More RESTful URLs.

new-endpoints

### CODE REVIEW

**Review by joeldalley — APPROVED** [review]
Looks good, and as I understand it, everyone agrees on including the supplier ID in all paths. Are we ready to merge this?

**Review by jaymichls — APPROVED** [review]

File: docs/openapi_v1.0.0-alpha.yaml [hunk 1]
```diff
@@ -202,18 +211,19 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
-  /reservations:
-    post:
+  /suppliers/{supplierId}/bookings:
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml:214]: This endpoint definition is missing a placeholder for the `uuid` path parameter called out in the definition below.

```suggestion
  /suppliers/{supplierId}/bookings/{uuid}:
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 2]
```diff
@@ -274,17 +288,18 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/InternalServerError'
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml:291]: This looks unnecessary as it is a duplicate of the `getBooking` endpoint above. Both endpoints have the same request method and call for 2 path parameters: `supplierId` and `uuid`. I suggest removing this definition and correcting the earlier definition since it has the correct `operationId`.

File: docs/openapi_v1.0.0-alpha.yaml [hunk 3]
```diff
@@ -1204,7 +1263,7 @@ components:
         format: string
     Uuid:
       name: 'uuid'
-      in: query
+      in: path
       description: |
         A valid booking reservation UUID that matches the `uuid` sent during the initial `POST /reservations`
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: Need to update this description to match the new endpoint structure.

```suggestion
         A valid booking reservation UUID that matches the `uuid` sent during the initial `POST /suppliers/{supplierId}/bookings`
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 4]
```diff
@@ -347,15 +367,20 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/InternalServerError'
-    put:
+
+  /suppliers/{supplierId}/bookings/{uuid}/cancel/confirm:
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: Nesting actions like this sets a bad pattern. Let's shorten this to a single action.

```suggestion
  /suppliers/{supplierId}/bookings/{uuid}/confirmCancel:
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 5]
```diff
@@ -516,20 +575,20 @@ components:
           description: |
             This is the RECOMMENDED refresh interval for the Reseller and SHOULD be used by the Reseller to control the frequency at which they make a `getBooking` request for the following scenarios:
 
-            * To see if a booking/cancellation has changed out of a `PENDING` status into `CONFIRMED` or `REJECTED`.
+            * To see if a booking/cancel has changed out of a `PENDING` status into `CONFIRMED` or `REJECTED`.
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: I think this may have been mistakenly shortened. This refers to either a Booking or Cancellation request.

```suggestion
            * To see if a booking/cancellation has changed out of a `PENDING` status into `CONFIRMED` or `REJECTED`.
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 6]
```diff
@@ -347,15 +367,20 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/InternalServerError'
-    put:
+
+  /suppliers/{supplierId}/bookings/{uuid}/cancel/confirm:
+    post:
       tags:
-        - Cancellations
+        - Bookings
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: Same as above.

```suggestion
        - Cancellations
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 7]
```diff
@@ -326,9 +344,11 @@ paths:
           $ref: '#/components/responses/InternalServerError'
     delete:
       tags:
-        - Cancellations
+        - Bookings
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: Same as above.

```suggestion
        - Cancellations
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 8]
```diff
@@ -299,16 +314,19 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
-  /cancellations:
+  /suppliers/{supplierId}/bookings/{uuid}/cancel:
     post:
       tags:
-        - Cancellations
+        - Bookings
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml]: This should still be tagged as a Cancellation. The tag doesn't refer to the path, rather it's a way to group endpoints based on the type of operation you're trying to do. In this case, the endpoint is only applicable to a Cancellation.

```suggestion
        - Cancellations
```

File: docs/openapi_v1.0.0-alpha.yaml [hunk 9]
```diff
@@ -1415,6 +1462,4 @@ tags:
   - name: Suppliers
   - name: Products
   - name: Availability
-  - name: Reservations
-  - name: Cancellations
```
Comment (gimpster) [inline docs/openapi_v1.0.0-alpha.yaml:1419]: Going along with the above tagging, this should be retained in order to display the tag grouping correctly.

```suggestion
  - name: Cancellations
```

**Review by gimpster — CHANGES_REQUESTED** [review]
Looks good overall, just a few cleanup items.

**Comment by trojkat** [comment]
@gimpster thanks for the review, please recheck. I've also updated the screenshot.

**Comment by ollym** [comment]
Need this merged before I can do my cancellation changes as there would be too many conflicts. @gimpster can you do this please?

**Review by gimpster — APPROVED** [review]
Looks good. Merging.
